### PR TITLE
chore: Split long running integration tests and fast unit tests (DEV-1537)

### PR DESCRIPTION
**Pull request by seakayone**
### Task Description/Number
Split tests into different source sets, so that long running integration tests are separated from faster unit tests. 

Make `sbt it` run all tests from all sbt sub projects from the **IntegrationTest configuration in sequence**.

Make `stb test` run all tests from all sbt sub projects from the **Test configuration in parallel**.

Run integration and unit tests in parallel Github Action build jobs.

Move most tests to `it` source set and  keep some zio-test unit tests in test source set.

Fix schema-core to run tests by adding zio-test-sbt to its dependencies.

Add documentation: README.md and internal docs.

Issue Number: [DEV-1537](https://dasch.atlassian.net/browse/DEV-1537)

### Basic Requirements

Please check if your PR fulfils the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### PR Type

What kind of change does this PR introduce?

- [ ] Bugfix: represents bug fixes
- [ ] Refactor: represents production code refactoring
- [ ] Feature: represents a new feature
- [ ] Documentation: documentation changes (no production code change)
- [ ] Chore: maintenance tasks (no production code change)
- [ ] Style: styles updates (no production code change)
- [x] Test: all about tests: adding, refactoring tests (no production code change)
- [ ] Other... Please describe:

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### Does this PR change client-test-data?

- [ ] Yes (don't forget to update the JS-LIB team about the change)
- [x] No

### CODE REVIEW

File: dsp-schema/core/src/test/scala/dsp/schema/domain/SchemaCommandsSpec.scala [hunk 1]
```diff
@@ -11,7 +11,7 @@ import dsp.valueobjects.LanguageCode
 import dsp.valueobjects.Schema
 
 /**
- * This spec is used to test [[dsp.schema.domain.SchemaCommands]].
+ * This spec is used to test [[dsp.schema.domain.CreatePropertyCommand]].
```
Comment (irinaschubert) [inline dsp-schema/core/src/test/scala/dsp/schema/domain/SchemaCommandsSpec.scala:14]:
```suggestion
 * This spec is used to test [[dsp.schema.domain.SchemaCommandsSpec]].
```

**Review by irinaschubert — APPROVED** [review]
LGTM

File: webapi/src/test/scala/org/knora/webapi/messages/v2/responder/ontologymessages/KnoraCardinalityInfoIsStricterThanZSpec.scala [hunk 2]
```diff
@@ -1,13 +1,12 @@
-/*
```
[the remark this answers is not in the document]
Comment (seakayone) [reply webapi/src/test/scala/org/knora/webapi/messages/v2/responder/ontologymessages/KnoraCardinalityInfoIsStricterThanZSpec.scala:1]: I agree, this should be automated though.

[the remark this answers is not in the document]
Comment (seakayone) [reply webapi/src/test/scala/org/knora/webapi/messages/v2/responder/ontologymessages/KnoraCardinalityInfoIsStricterThanZSpec.scala:1]: I remember that some build tool plugin could do this work for us.

[the remark this answers is not in the document]
Comment (seakayone) [reply webapi/src/test/scala/org/knora/webapi/messages/v2/responder/ontologymessages/KnoraCardinalityInfoIsStricterThanZSpec.scala:1]: https://github.com/sbt/sbt-header e.g.

[the remark this answers is not in the document]
Comment (irinaschubert) [reply webapi/src/test/scala/org/knora/webapi/messages/v2/responder/ontologymessages/KnoraCardinalityInfoIsStricterThanZSpec.scala:1]: sounds good :)

File: build.sbt [hunk 3]
```diff
@@ -27,7 +24,8 @@ lazy val aggregatedProjects: Seq[ProjectReference] = Seq(webapi, sipi)
 
 lazy val buildSettings = Seq(
   organization := "org.knora",
-  version      := (ThisBuild / version).value
+  version      := (ThisBuild / version).value,
+  javaOptions += "-Xmx2G"
```
[the remark this answers is not in the document]
Comment (seakayone) [reply build.sbt:28]: The OOM exception did not occur anymore. I will remove this setting here.

File: webapi/src/test/scala/org/knora/webapi/messages/v2/responder/ontologymessages/KnoraCardinalityInfoIsStricterThanZSpec.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (seakayone) [reply webapi/src/test/scala/org/knora/webapi/messages/v2/responder/ontologymessages/KnoraCardinalityInfoIsStricterThanZSpec.scala]: => https://github.com/dasch-swiss/dsp-api/pull/2317

**Pull request merged**